src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -518,10 +518,13 @@ private void WriteTo(AnalyzerReference reference, ObjectWriter writer, bool chec
                 writer.WriteString(nameof(AnalyzerFileReference));
                 writer.WriteInt32((int)SerializationKinds.FilePath);
 
-                if (!checksum)
+                if (checksum)
                 {
                     // we don't write full path when creating checksum
-                    writer.WriteString(file.FullPath);
+                    // make sure we always normalize assemblyPath to lower case since it comes from Assembly.Location
+                    // unlike FullPath which comes from string
+                    writer.WriteString(assemblyPath.ToLowerInvariant());
```

Comment:
hmmm.. @tannergooding that probably need to be fixed next time - opened tracking issue https://github.com/dotnet/roslyn/issues/16447

for now, since that path is always point to shadow copied path which guarantee unique, the way it does now, shouldn't break anything.